### PR TITLE
Implement point two of #216 with tests

### PR DESCRIPTION
**Pull request by alshakero**
Fixes point 2 of #216

### CODE REVIEW

File: src/palindrom-dom.js [hunk 1]
```diff
@@ -79,6 +78,30 @@ const PalindromDOM = (() => {
                 this.morphUrlEventHandler
             );
         }
+        
+        /**
+         * @param {String} href
+         * @throws {Error} network error if occured
+         */
+        async getPatchUsingHTTP(href) {
```
Comment (tomalec) [inline src/palindrom-dom.js]: I don't see why it's async, it does not await for a thing.

File: src/palindrom-dom.js [hunk 2]
```diff
@@ -79,6 +78,30 @@ const PalindromDOM = (() => {
...
+                },
+                cancelable: true,
+                bubbles: true
+            });
+            
+            this.element.dispatchEvent(event);
+
+            // check if event was canceled
+            if(event.returnValue) {
+                this.network.getPatchUsingHTTP(href);
```
Comment (tomalec) [inline src/palindrom-dom.js]: `await` missing here?

File: test/specs/dom/index.js [hunk 3]
```diff
@@ -183,7 +183,6 @@ if (typeof window !== 'undefined') {
 
                   setTimeout(function() {
                     setTimeout(() => { expect(historySpy.callCount).to.equal(1); done() });
-                    done();
```
Comment (tomalec) [inline test/specs/dom/index.js]: Could you break line 185, so the expect would have its own?

Comment (alshakero) [reply test/specs/dom/index.js]: I'll format the whole code later after 216 ends.

File: src/palindrom-dom.js [hunk 4]
```diff
@@ -79,6 +78,30 @@ const PalindromDOM = (() => {
                 this.morphUrlEventHandler
             );
         }
+        
+        /**
+         * @param {String} href
+         * @throws {Error} network error if occured
```
Comment (tomalec) [inline src/palindrom-dom.js]: Cosmetic: You can add `@fires` with event name and description

File: test/specs/dom/index.js [hunk 5]
```diff
@@ -178,14 +178,19 @@ if (typeof window !== 'undefined') {
...
 
-                  setTimeout(function() {
-                    setTimeout(() => { expect(historySpy.callCount).to.equal(1); done() });
-                    done();
-                  }, 5);
-                }, 5);
+                createAndClickOnLinkNestedShadowDOMContent();
+
+                setTimeout(function() {
+                  debugger
```
Comment (tomalec) [inline test/specs/dom/index.js]: Please remove `debugger`

**Pull request merged**